components/sendgrid/actions/delete-contacts/delete-contacts.js

```diff
@@ -0,0 +1,56 @@
+const sendgrid = require("../../sendgrid.app");
+const validate = require("validate.js");
+const common = require("../common");
+
+module.exports = {
+  key: "sendgrid-delete-contacts",
+  name: "Delete Contacts",
+  description: "Allows you to delete one or more contacts.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    sendgrid,
+    deleteAllContacts: {
+      type: "boolean",
+      label: "Delete All Contacts?",
+      description: "This parameter allows you to delete all of your contacts. This can not be used with the `Ids` parameter.",
+      optional: true,
+    },
+    ids: {
+      type: "object",
+      label: "Ids",
+      description: "An array of contact IDs to delete.",
+      optional: true,
```

Comment:
An object here makes no sense IMO
![image](https://user-images.githubusercontent.com/5479513/126414537-da33a08d-246d-499e-998f-752bf490b197.png)

The way to inject values into that prop is by setting its type to `string` (as discussed offline in Slack), and do the runtime validation during the action execution to make sure that the injected value complies with the expected type/format. In other words, making this a prop of type `string` allows users to specify a reference to another value (which will then be injected by the platform) and it is this last value the one that we must validate.

```suggestion
      type: "string",
      label: "Contact ID's",
      description: "An array of contact IDs to delete",
      optional: true,
```

Reply:
Ok, I had used object instead of string when receiving arrays to that we could leverage the validation capabilities from the `validate.js` library. Having the string we will have to parse and maybe use custom validator under `validate.js`